### ISSUE TITLE
Add #code_native #code_llvm and #code_typed

### ISSUE THREAD

**Issue by wolfv** [QuantStack/xeus-cling#193]
Borrowing from Julia here. These are pretty cool macros to get the assembly, LLVM bytecode or AST.

We could introduce them as cell magics. 

Would just be necessary to find a way to extract that info in a readable way from cling.

**Comment by wolfv** [comment]
do you have any idea if the cling guys have tools for this already? Or who to ask?

**Comment by SylvainCorlay** [comment]
I don't think so. We have the ClingValue for the last returned value handy, but not the whole cell byte code.

**Comment by wolfv** [comment]
We've got an "official" endorsement from Axel on the cling forums for that: https://root-forum.cern.ch/t/get-assembly-llvm-bytecode-or-clang-ast-output/30800

**Comment by wolfv** [comment]
I've started an exploratory commit here: https://github.com/root-project/cling/pull/262 

Hoping to get some feedback from the cling folks.
